### DISCUSSION TITLE
Page Number Navigation

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#284] [Component suggestions]
<img width="480" alt="visuele weergave van een page number navigation component" src="https://github.com/user-attachments/assets/61fad6e2-2c86-4c44-82cd-e2283db442de">

We willen graag meer leren over de Page Number Navigation component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1727676121007699)

## Naam
Page Number Navigation

## Doel
Navigatie naar andere pagina's wanneer een grote hoeveelheid informatie is verdeeld over genummerde pagina's.

## Ook bekend als
Pagination, Paginator

## Onderzoek
In een [comment bij GOV UK's Pagination Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/77#issuecomment-827500041) wordt aangegeven dat het gebruik van paginanummers in plaats van alleen “vorige” en “volgende” knoppen de gebruiksvriendelijkheid aanzienlijk heeft verbeterd, waardoor alle 105 geteste gebruikers de taak succesvol konden voltooien.

In een [comment bij GOV UK's Pagination Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/77#issuecomment-1576806517) wordt aangegeven dat de labels ‘volgende’ en ‘vorige’ meer context nodig hebben, zoals ‘volgende pagina’, zodat gebruikers van schermleestechnologieën beter begrijpen wat het doel van de links is wanneer deze buiten hun context worden bekeken.

[Pagination Best Practices - Uxcel](https://app.uxcel.com/courses/ui-components-best-practices/best-practices-005)
In dit artikel worden best practices besproken, zoals het beperken van overmatige paginalinks, zorgen dat knoppen groot genoeg zijn om gemakkelijk aan te klikken, en het duidelijk markeren van de huidige pagina.

[Alternatives to Pagination on Product-Listing Pages](https://arc.net/l/quote/naeniaxa)
In dit artikel wordt verteld dat infinite scrolling gebruikers frustreert doordat de footer onbereikbaar wordt omdat deze steeds verder naar beneden wordt verschoven wanneer nieuwe items automatisch worden toegevoegd.

[Infinite Scrolling, Pagination Or “Load More” Buttons? Usability Findings In eCommerce - Smashing Magazine](https://www.smashingmagazine.com/2016/03/pagination-infinite-scrolling-load-more-buttons/)
In dit artikel bespreekt Christian Holst de gebruikersonderzoeksresultaten van het Baymard Institute, waarin wordt aangetoond dat de combinatie van “Load more”-knoppen en lazy-loading een betere gebruikerservaring biedt dan traditionele paginering of infinite scrolling van toonaangevende **e-commercewebsites**.​

[Accessible Components: The Pagination - LinkedIn](https://www.linkedin.com/pulse/accessible-components-pagination-roc%C3%ADo-alvarado/)
In dit artikel biedt Rocio technieken om paginering toegankelijker te maken voor alle gebruikers, met een focus op keyboardinteracties en interacties met schermlezers.

**Comment by Rozerinay** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-navigation-pagination--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29206&node-type=canvas&t=RnwuBh3WiDJV1MwF-11)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/pagination)

---

### Screenshots / Varianten

**Storybook**

First page
<img width="400" alt="image" src="https://github.com/user-attachments/assets/d527c8d2-b992-4406-a24d-104fe0b4cf74"><br>

Second page
<img width="480" alt="image" src="https://github.com/user-attachments/assets/75f1a20e-55f7-495f-ba35-8085eec9ecf9"><br>

Last page
<img width="400" alt="image" src="https://github.com/user-attachments/assets/0a2b218a-5263-4ee9-9a26-3799968bb0f0">

Responsive gedrag (bij verkleinen scherm)
<img width="297" alt="image" src="https://github.com/user-attachments/assets/8be69534-3034-4c3a-a944-8d16fdf14565"><br>

**Figma**

First page
<img width="560" alt="image" src="https://github.com/user-attachments/assets/cdc15a69-33ac-4513-b152-0e8953da6443">

Second page
<img width="560" alt="image" src="https://github.com/user-attachments/assets/0b4c326d-a315-47f7-86ea-2871f74484e5">

Third page
<img width="560" alt="image" src="https://github.com/user-attachments/assets/986ecbca-b957-45c2-a718-f065f6a29db4">

Middle page
<img width="560" alt="image" src="https://github.com/user-attachments/assets/1fd1f61d-918a-49ef-aefe-5dbbd2774567">

Second for last page
<img width="560" alt="image" src="https://github.com/user-attachments/assets/b732d382-b26d-4459-8f1f-ccac6a3c6dfc">

For last page
<img width="560" alt="image" src="https://github.com/user-attachments/assets/2fded785-d829-4b90-a178-3e9fbd31c7b4">

Last page
<img width="560" alt="image" src="https://github.com/user-attachments/assets/3e259b08-f790-4ca9-9937-945d4910f3be">


---

### Notities
- Onder de naam: Pagination.
- Button met label + icon wordt gebruikt voor 'Vorige' en 'Volgende'. 
- Buttons krijgen underline bij hover.
- Indicatie pagina: huidige paginanummer is bold en de overige pagina's zijn regular.
- Als je op de eerste of laatste pagina bent, heb je slechts één Button: Vorige of Volgende, afhankelijk van je positie.
- Ellipsis wordt gebruikt voor tussenliggende pagina's. Is niet aanklikbaar.
- Aantal zichtbare paginanummers: maximaal 6 in Storybook en 5 in Figma.
- Responsive gedrag: in Storybook komt de 'Volgende' button op de volgende regel te staan.
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/react-navigation-pagination--docs)
- [Figma - Huidig](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=1673-231&node-type=frame&t=1VhtrKIdzgsxtTIT-11)
- [Figma - WIP](https://www.figma.com/design/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?node-id=1691-11779&node-type=canvas&t=OVJ4boNFcgIK5O3m-11)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Pagination)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="300" alt="image" src="https://github.com/user-attachments/assets/c22f7eaf-3130-48ee-88cf-061960696d9f">

Disabled
<img width="300" alt="image" src="https://github.com/user-attachments/assets/3b60131c-3560-47ae-9131-023618f42d49">

Four items
<img width="300" alt="image" src="https://github.com/user-attachments/assets/74119922-b2f7-41fc-84b9-4dec8e015b98">

Five items
<img width="300" alt="image" src="https://github.com/user-attachments/assets/e6c38638-4007-4237-9168-21610af52243">

Aligned start
<img width="340" alt="image" src="https://github.com/user-attachments/assets/7633b9c2-bbd6-43d2-ab2f-cbac6bb317dc">

Aligned center
<img width="340" alt="image" src="https://github.com/user-attachments/assets/d2ff0d10-8c12-459c-b29b-9bd03fd11330">

Aligned end
<img width="340" alt="image" src="https://github.com/user-attachments/assets/5274be68-a4f8-4907-aa5c-8aed82e66863">

**Figma - Huidig**
<img width="400" alt="image" src="https://github.com/user-attachments/assets/eab13eae-b798-46c6-b722-dd0953a5f139">

---

### Notities
- Onder de naam: Pagination.
- Icon (chevron) wordt gebruikt voor 'Vorige' en 'Volgende'.
- Buttons krijgen een andere kleur bij hover.
- Indicatie pagina: huidige paginanummer is bold en de overige pagina's zijn regular.
- Als je op de eerste of laatste pagina bent, is Vorige of Volgende icon disabled, afhankelijk van je positie.
- Ellipsis wordt gebruikt voor tussenliggende pagina's. Is niet aanklikbaar.
- Aantal zichtbare paginanummers: maximaal 5.
- Responsive gedrag: gehele Pagination blijft op 1 regel staan.
- Positie: biedt opties voor de plaatsing van de Pagination aan de linkerkant, in het midden of aan de rechterkant van de pagina.
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/62.8.4/components/layout/pagination
)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-pagination--pagination)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/pagination)

---

### Screenshots / Varianten

**Storybook**

First page
<img width="280" alt="image" src="https://github.com/user-attachments/assets/205cc7a2-08f6-4605-9b5a-c89316b40299">

Second page
<img width="280" alt="image" src="https://github.com/user-attachments/assets/269da302-5e2e-4c41-8336-d078b6dfd77d">

Fourth page
<img width="280" alt="image" src="https://github.com/user-attachments/assets/aac9f7aa-e67b-4b58-a726-c421475a10df">

For last page
<img width="280" alt="image" src="https://github.com/user-attachments/assets/321875cd-e33f-40cd-95c0-6230c0b22481">

Last page
<img width="280" alt="image" src="https://github.com/user-attachments/assets/805eaaf0-93d7-4a39-874a-21e77c03757a">

---

### Notities
- Onder de naam: Pagination.
- Icon (chevron) wordt gebruikt voor 'Vorige' en 'Volgende'.
- Buttons krijgen een underline kleur bij hover.
- Indicatie pagina: huidige pagina is omcirkeld.
- Als je op de eerste of laatste pagina bent, heb je slechts één Button: Vorige of Volgende icon, afhankelijk van je positie.
- Er wordt géén gebruik gemaakt van ellipsis.
- Aantal zichtbare paginanummers: maximaal 5.
- Eerste en laatste pagina is altijd zichtbaar.
- Responsive gedrag: gecentreerde overflow wrap.

**Comment by Rozerinay** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/pagination)

---

### Screenshots / Varianten

<img width="401" alt="image" src="https://github.com/user-attachments/assets/d863b652-6589-42d4-a4af-92ad4ef86aaa"><br>

Werking t/m 6 pagina's
<img width="580" alt="image" src="https://github.com/user-attachments/assets/e648f8d5-83ef-4858-84f5-78ebb62f2960">

Werking meer dan 6 pagina's
<img width="580" alt="image" src="https://github.com/user-attachments/assets/eb2fd17c-7fa4-4b5f-aa55-1f3af9cb3528">

Responsive gedrag
<img width="480" alt="image" src="https://github.com/user-attachments/assets/78dc76cb-d9c5-4f0e-be98-082d015c591e">


---

### Notities
- Onder de naam: Pagination.
- Button met label + icon wordt gebruikt voor 'Vorige' en 'Volgende'.
- Buttons krijgen een andere achtergrondkleur bij hover.
- Indicatie pagina: huidige paginanummer heeft een achtergrondkleur.
- Als je op de eerste of laatste pagina bent, heb je slechts één Button: Vorige of Volgende, afhankelijk van je positie.
- Ellipsis wordt gebruikt voor tussenliggende pagina's (meer dan 6). Is niet aanklikbaar.
- Aantal zichtbare paginanummers: maximaal 6.
- Responsive gedrag: gehele Pagination blijft op 1 regel staan. Huidige pagina en laatste pagina zijn dan altijd zichtbaar.
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## OpenGemeenten

[PatternLab](https://patternlab.lenteveld.nl/patterns/01-molecules-020-Card-Lists-30-Pagination-OpenGemeentenPagination-Pagination/01-molecules-020-Card-Lists-30-Pagination-OpenGemeentenPagination-Pagination.html)

---

### Screenshots / Varianten

Pagination
<img width="450" alt="image" src="https://github.com/user-attachments/assets/30ecd112-7e33-4a97-a6f2-613e34a53bb4">

Pagination next
<img width="450" alt="image" src="https://github.com/user-attachments/assets/2d143c44-239a-46a9-bd45-d819fbcdf856">

Pagination previous
<img width="450" alt="image" src="https://github.com/user-attachments/assets/33d873c4-e81f-48d3-9669-feee7b0af497">

---

### Notities
- Onder de naam: Pagination.
- Button met label + icon wordt gebruikt voor 'Vorige' en 'Volgende'.
- Geen paginanummers weergegeven. In plaats daarvan wordt in 1 regel de huidige en laatste pagina aangegeven. 
- Als je op de eerste of laatste pagina bent, is Vorige of Volgende disabled, afhankelijk van je positie.
- Responsive gedrag: gehele Pagination blijft op 1 regel staan en 'Vorige' en 'Vorige' veranderen naar een arrow icon.

**Comment by Rozerinay** [comment]
## Rotterdam

[Figma](https://www.figma.com/design/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?node-id=16303-10877&node-type=frame&t=jl7nU0kW2g3jRAHU-11)

---

### Screenshots / Varianten

Pagination with only 3 pages (page 2 is active)
<img width="286" alt="image" src="https://github.com/user-attachments/assets/a98dea15-7221-4785-a9ce-359ab0540ca5">

Pagination with more than 3 pages (page 2 is active)
<img width="286" alt="image" src="https://github.com/user-attachments/assets/bdbb0e79-86ed-4f9b-b594-eab5fc42b506">

Variant explanation
<img width="203" alt="image" src="https://github.com/user-attachments/assets/0356c805-8af0-448a-8ff7-4a975c969c76">

---

### Notities
- Onder de naam: Pagination.
- Icon (chevron) wordt gebruikt voor 'Vorige' en 'Volgende'.
- Buttons krijgen een andere achtergrondkleur bij hover.
- Indicatie pagina: huidige paginanummer is bold en heeft een achtergrondkleur.
- Ellipsis wordt gebruikt voor tussenliggende pagina's (meer dan 4). Is niet aanklikbaar.
- Aantal zichtbare paginanummers: maximaal 4.
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/pagination/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-page-number-navigation--documentatie)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=220-4774&node-type=canvas&t=Z3e12Cj7yLNDKplr-11)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/pagination)

---

### Screenshots / Varianten

**Storybook**

3 pages, first page active
<img width="220" alt="image" src="https://github.com/user-attachments/assets/777e3548-dc31-4931-ae10-44c8e992d8af">

5 pages, first page active
<img width="220" alt="image" src="https://github.com/user-attachments/assets/d97fca44-f0a0-4f08-82e0-78fc1ceb86f5">

30 pages, first page active
<img width="220" alt="image" src="https://github.com/user-attachments/assets/c405b134-8389-4272-a198-36120ea50082">

30 pages, second page active
<img width="374" alt="image" src="https://github.com/user-attachments/assets/0b9ccd0a-8530-4c7a-94cb-889072321125">

30 pages, fourth page active
<img width="431" alt="image" src="https://github.com/user-attachments/assets/2a7d8e58-59e0-48f0-b662-0452e922c752">

30 pages, fifth page active
<img width="429" alt="image" src="https://github.com/user-attachments/assets/2638f67a-27bc-415a-8408-0e488733c4b1">

Responsive gedrag
<img width="244" alt="image" src="https://github.com/user-attachments/assets/d63edaa4-7e89-4550-a4fb-8b098727c6ea">

**Figma**
<img width="331" alt="image" src="https://github.com/user-attachments/assets/fda3b22e-d851-42fc-901c-d49835184e3e">

---

### Notities
- Onder de naam: Pagination.
- Button met label + icon wordt gebruikt voor 'Vorige' en 'Volgende'.
- Buttons krijgen een andere achtergrondkleur en worden bold bij hover.
- Indicatie pagina: huidige paginanummer is bold.
- Als je op de eerste of laatste pagina bent, heb je slechts één Button: Vorige of Volgende, afhankelijk van je positie.
- Ellipsis wordt gebruikt voor tussenliggende pagina's (meer dan 3). Is niet aanklikbaar.
- Aantal zichtbare paginanummers: maximaal 6.
- Responsive gedrag: gecentreerde overflow wrap. Volgende button komt op volgende regel te staan.
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-pagination--docs)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=1691-11779&node-type=canvas&t=gvvif35zxodbcKvU-11)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/pagination)

---

### Screenshots / Varianten

**Storybook**

<img width="374" alt="image" src="https://github.com/user-attachments/assets/83a9efe1-cbd9-47d5-a51e-a2c54739c937">

Responsive gedrag
<img width="274" alt="image" src="https://github.com/user-attachments/assets/37e77fae-e89a-42cd-abf2-d9003ccbd8ee">

**Figma**
<img width="600" alt="image" src="https://github.com/user-attachments/assets/d2c3a83b-2842-463c-a972-a2ce83e43a89">

**Publieke website**

First page active
<img width="350" alt="image" src="https://github.com/user-attachments/assets/add78b76-d00e-44d7-bb19-dd0c4f279f99">

Last page active
<img width="350" alt="image" src="https://github.com/user-attachments/assets/d498129d-32d5-402d-9191-9c0780eeb492">

More than 5 pages
<img width="400" alt="image" src="https://github.com/user-attachments/assets/84aa542a-9b33-4412-b6ad-64f9f421e1c3">

---

### Notities
- Onder de naam: Pagination.
- Verschillende versies in Figma. Uitwerking in Figma en Storybook lopen uiteen. Onderstaande notities gaan uit van de uitwerking in Storybook.
- Button met label wordt gebruikt voor 'Vorige' en 'Volgende'.
- Buttons krijgen een andere achtergrondkleur bij hover.
- Indicatie pagina: huidige paginanummer heeft een achtergrondkleur.
- Als je op de eerste of laatste pagina bent, heb je slechts één Button: Vorige of Volgende, afhankelijk van je positie.
- Geen gebruik van ellipsis.
- Aantal zichtbare paginanummers: maximaal 5.
- Responsive gedrag: overflow wrap. Volgende button komt op volgende regel te staan.

**Comment by Marwaxhello** [comment]
## Developer Overheid

- [Documentatie website](https://don.apps.digilab.network/)
- [GitHub](https://github.com/developer-overheid-nl})
- [Website (page met page-number-navigation)](https://developer.overheid.nl/apis)

---

### Screenshots / Varianten
<img width="500" alt="Screenshot 2025-02-11 at 14 39 38" src="https://github.com/user-attachments/assets/e1c5c150-4eff-41c7-abc9-eb5c9c5911dd" />


**Page Number Navigation**


### Notities

- Onder de naam: Pagination list.
- Button met icon wordt gebruikt voor 'Vorige' en 'Volgende'.
- Buttons krijgen een andere achtergrondkleur bij hover.
- Indicatie pagina: huidige paginanummer heeft een achtergrondkleur.

**Comment by Robbert** [comment]
## Ontdek Tynaarloo

['t Noordelijk Puntje op de kaart van Ontdek Tynaarloo](https://ontdek.tynaarlo.nl/kaart/t-noordelijk-puntje-ba44954?l=default#11.4/53.1747/6.5296)

## Screenshots / varianten

In een Dialog met details over een locatie op de kaart, staan een "Previous" en "Next" Button om een andere locatie op de kaart te bekijken. De navigatie is erg compact, en biedt alleen de mogelijkheid om de vorige en volgende item te bekijken. Daar tussen in staat bijvoorbeeld de tekst "1 van 2".

Detail:

<img width="133" alt="Screenshot 2025-03-06 at 13 26 26" src="https://github.com/user-attachments/assets/b9b457ae-2948-42ee-a213-10491f40f44f" />

In de context:

<img width="1332" alt="Screenshot 2025-03-06 at 13 29 09" src="https://github.com/user-attachments/assets/166eeb1a-e06e-4e99-a2aa-bf85e0939187" />

**Comment by Rozerinay** [comment]
Ik zei eerder in een comment in de NL Design System Figma bibliotheek:

> Over het algemeen voelen gebruikers zich het meest comfortabel met een patroon dat aansluit bij hun behoefte op dat moment. Oneindig scrollen wordt als snel en makkelijk ervaren voor vrijblijvende verkenning, maar kan onprettig worden bij een doelgerichte zoektocht door gebrek aan overzicht.
> 
> Paginering geeft structuur en rustpunten, maar voelt traag in situaties waarin men juist veel wil ontdekken. De ‘Load more’-knop wordt vaak gezien als gebruiksvriendelijk compromis: het laat mensen in hun eigen tempo meer content laden, wat zowel de perceptie van snelheid als het gebruiksgemak ten goede komt.
> 
> Welke methode “het beste” is, hangt dus sterk af van de gebruikerscontext. Een zorgvuldige keuze – en soms een combinatie – van deze patronen kan ervoor zorgen dat de interface zowel snel aanvoelt, eenvoudig te gebruiken is, als tegemoetkomt aan de voorkeuren van de gebruiker.

**Bronnen**
De bovenstaande bevindingen zijn gebaseerd op gebruikersonderzoek en praktijkcases uit o.a. [Baymard Institute](https://www.smashingmagazine.com/2016/03/pagination-infinite-scrolling-load-more-buttons/#:~:text=Throughout%20our%20large,the%20first%20page%20of%20results), [Nielsen Norman Group](https://www.nngroup.com/articles/infinite-scrolling-tips/#:~:text=Because%20of%20the%20usability%20issues,the%20listing%20page%E2%80%99s%20content%20to) en UX-experts, evenals cases van platforms (bijv. [Etsy’s experiment](https://danwin.com/2013/01/infinite-scroll-fail-etsy/#:~:text=%E2%80%9CSeeing%20more%20items%20faster%20is,%E2%80%9D)).

Deze bronnen tonen consistent aan dat elk patroon situaties kent waar het uitblinkt, en dat de perceptie van snelheid en gebruiksgemak uiteindelijk samenhangt met hoe goed het patroon past bij de doelen van de gebruiker.

**Comment by eslook** [comment]
## Verduidelijking van de gekozen naamgeving van "Pagination" naar "Page Number Navigation"

In [het Help Wanted draadje op Slack is op 16 oktober 2024](https://codefornl.slack.com/archives/C7KDQRXMZ/p1729095858695049?thread_ts=1728994500.695589&cid=C7KDQRXMZ) verduidelijkt dat de de naam "Page Number Navigation" gekozen is om de volgende redenen:

- ook duidelijk in automatische vertalingen
- bestaat helemaal uit keywords die in webstandaarden voorkomen
- geen jargon, ook duidelijk voor niet technische gebruikers

"Pagination" en "Paginator" blijven bestaan als alias in de documentatie.
